packages/@lwc/engine-core/src/framework/stylesheet.ts

```diff
@@ -132,6 +133,18 @@ function evaluateStylesheetsContent(
                 // the stylesheet, while internally, we have a replacement for it.
                 stylesheet = getStyleOrSwappedStyle(stylesheet);
             }
+
+            // Check that this stylesheet was generated by our compiler
+            if (!isStylesheetRegistered(stylesheet)) {
+                logError(
+                    `TypeError: Unexpected LWC stylesheet content found for component <${vm.tagName.toLowerCase()}>.`
+                );
+                report(ReportingEventId.UnexpectedStylesheetContent, {
+                    tagName: vm.tagName.toLowerCase(),
+                });
+                continue;
```

Comment:
This `continue` makes this a functional (breaking) change. My understanding was that we were just going to warn, but still call the stylesheet function?

Reply:
Also, let's log a GH issue to track converting the warning into an error, then mention it with a `TODO` comment here. We can add it to the breaking changes wishlist: #2964

Reply:
My bad, that detail slipped my mind. :+1: on logging a GH issue to convert the warning to error.

Reply:
To play it safe, can we also add a test to check that the CSS string returned by the arbitrary function is actually inserted into the DOM? That would have caught the `continue`.